lib/init.js

```diff
@@ -15,17 +17,26 @@ init.usage = usage(
   'init',
   '\nnpm init [--force|-f|--yes|-y|--scope]' +
   '\nnpm init <@scope> (same as `npx <@scope>/create`)' +
-  '\nnpm init [<@scope>/]<pkg> (same as `npx [<@scope>/]create-<pkg>`)'
+  '\nnpm init [<@scope>/]<name> (same as `npx [<@scope>/]create-<name>`)' +
+  '\nnpm init <pkg> (same as `npx <pkg>`)'
 )
 
 function init (args, cb) {
   if (args.length) {
     var NPM_PATH = path.resolve(__dirname, '../bin/npm-cli.js')
     var initerName = args[0]
-    var packageName = /^@[^/]+$/.test(initerName)
-      ? initerName + '/create'
-      : initerName.replace(/^(@[^/]+\/)?/, '$1create-')
-
+    var packageName = initerName
+    if (/^@[^/]+$/.test(initerName)) {
+      packageName = initerName + '/create'
+    } else {
+      var req = npa(initerName)
+      if (isRegistry(req)) {
```

Comment:
I wonder if we could do more here. Should we also, maybe, handle hosted git deps? `npm init zkat/templ` -> `npx github:zkat/create-templ`? What's the expectation around this? What makes these others any better than doing `npx file:some/initializer`?

Reply:
At the moment they aren't any better than using `npx`. We could keep it simple and when a user provides a non registry name `npm init` could recommend using `npx` instead. Something like
```shell
$ npm init file:some/initializer
> Unrecognized initializer: file:some/initializer.
> For more package binary executing power check out `npx`:
> https://www.npmjs.com/package/npx

Reply:
Ok, I've updated the PR to more gracefully punt on git, path, and url names for now instead of just erring out.